### PR TITLE
26 add gmail service

### LINKED ISSUE #26: Add Gmail service
Add the Gmail service to the backend

Allow to send email, or detect when receiving one, firstly on a simple app, and then integrated to the project.

### PR DESCRIPTION
**Pull request by GuyomT**
# Purpose

The purpose of this pull request is to add the service gmail.

Fixes #26 

## Added features

* This service has been added to the application. It's include:
  * Sending an email to any person with the mail aequallsquared@gmail.com created for this project.

### CODE REVIEW

File: server/services/gmail/gmail.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM_EMAIL = 'aequallsquared@gmail.com'
+
+const {google} = require('googleapis');
+
+async function sendEmail(to, subject, body) {
```
Comment (LucasTesnier) [inline server/services/gmail/gmail.js]: in convention async is set after the function name.

File: server/services/gmail/gmail.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const base64EncodedEmail = new Buffer.from(message).toString('base64');
+  const request = {
+    auth: auth,
+    userId: 'me',
+    resource: {
+      raw: base64EncodedEmail
+    }
+  };
+
+  console.log(`Email sent to ${to}!`);
```
Comment (LucasTesnier) [inline server/services/gmail/gmail.js]: Is it planned to be deleted in a future pr?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
+ Can you rename your gmail.js file to service.js or service_init.js ?
+ Did you format your JS Code with the VS extension ?

File: server/services/gmail/send_email.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const gmail = require('./gmail_init').getGmailClient();
+
+/**
+ *
```
Comment (LucasTesnier) [inline server/services/gmail/send_email.js]: Missing function brief

File: server/services/gmail/gmail_init.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM_EMAIL = 'aequallsquared@gmail.com'
+
+const {google} = require('googleapis');
+
+/**
+ *
```
Comment (LucasTesnier) [inline server/services/gmail/gmail_init.js]: Missing function brief

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just 2 minor documentation error, otherwise NTR (Check your format)

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**